contracts/src/v0.8/PermissionedForwardProxy.sol

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./ConfirmedOwner.sol";
+import {getRevertMsg} from "./utils/utils.sol";
+
+/**
+ * @title PermissionedForwardProxy
+ * @notice This proxy is used to forward calls from sender to target. It maintains
+ * a permission list to check which sender is allowed to call which target
+ */
+contract PermissionedForwardProxy is ConfirmedOwner {
+  mapping(address => address) public forwardPermissionList;
+
+  constructor() ConfirmedOwner(msg.sender) {}
+
+  /**
+   * @notice Verifies if msg.sender has permission to forward to target address and then forwards the handler
+   * @param target address of the contract to forward the handler to
+   * @param handler bytes to be passed to target in call data
+   */
+  function forward(address target, bytes calldata handler) external {
+    require(forwardPermissionList[msg.sender] == target, "Forwarding permission not found");
+    (bool success, bytes memory payload) = target.call(handler);
+    if (!success) {
+      revert(getRevertMsg(payload));
```

Comment:
Can you wrap this error somehow? Also a bit of a tangent... how does `getRevertMsg()` work when there is a custom error code? Does that get lost or do we capture it somehow?

Reply:
Yeah i can wrap this, this pattern is actually used here https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/upkeeps/CronUpkeep.sol#L82-L84 

Wasn't sure if the extra gas for wrapping error was worth it

Reply:
It gets lost I think. We return a hardocded error message if payload is small https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/utils/utils.sol#L9

Reply:
hmm with the addition of custom error messages, I'm worried that the `getRevertMsg` will actually obfuscate more than it reveals.

Reply:
I am not sure how custom error messages are returned in call data. The utility from openzepellin has this logic to extract revert reason https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/Address.sol#L201-L220 which seems to work for any return data

Reply:
I tested this with counter returning a custom error.

The openzepellin lilbrary copies the raw bytes and returns that custom error, while our function returns the string "transaction reverted silently". However it will be hard for clients of this contract to understand that custom error since there's no definition on this contract. 

I'll update this PR with that method since for revert strings the behaviour should remain same and copying custom error should be better than "transaction reverted silently"

Reply:
That sounds like a better solution to me